clang/lib/CIR/Dialect/IR/CIRDialect.cpp

```diff
@@ -2646,6 +2646,24 @@ ParseResult cir::InlineAsmOp::parse(OpAsmParser &parser,
   return mlir::success();
 }
 
+//===----------------------------------------------------------------------===//
+// ThrowOp
+//===----------------------------------------------------------------------===//
+
+mlir::LogicalResult cir::ThrowOp::verify() {
+  // For the no-rethrow version, it must have at least the exception pointer.
+  if (rethrows())
```

Comment:
Because you are adding the verifier you should also add a CIR to CIR tests (and ones for the invalid combos)